src/EventStore.Core/Authorization/LegacyStreamPermissionAssertion.cs

```diff
@@ -25,6 +30,20 @@ public class
 
 		public ValueTask<bool> Evaluate(ClaimsPrincipal cp, Operation operation, PolicyInformation policy,
 			EvaluationContext context) {
+			var startedAt = _sw.Elapsed;
+
+			if (!_allowAnonymousStreamAccess) {
+				var claims = cp.Claims;
+				
+				foreach (var claim in claims) {
+					if (claim.Type == ClaimTypes.Anonymous) {
```

Comment:
is it a bit odd to be checking for the existence of an anonymous claim rather than the absense of some other claim indicating that aren't anonymous